lib/exceptional_synchrony/event_machine_proxy.rb

```diff
@@ -62,6 +63,7 @@ def next_tick(&block)
     def stop
       @proxy_class.stop
       @proxy_class.next_tick { } #Fake out EventMachine's epoll mechanism so we don't block until timers fire
+      Thread.current.thread_variable_set(:running_em_synchrony, false)
```

Comment:
I wonder about using a name that captures that this is THE reactor thread? I guess that's implied by it being a thread local, but still I think that might get lost on some reading the code, for example where they might wonder about this setting for the deferred thread pool. Maybe something like `em_synchrony_reactor_thread = true`?

Reply:
Done